권지윤/Week 5/week5Study/app/src/main/java/com/jiyoxn/study/HumanAdapter.kt

```diff
@@ -0,0 +1,58 @@
+package com.jiyoxn.study
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.jiyoxn.study.databinding.ItemFriendBinding
+import com.jiyoxn.study.databinding.ItemMyBinding
+
+class HumanAdapter(context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private val inflater by lazy { LayoutInflater.from(context) }
+
+    val VIEW_TYPE_FRIEND = 0
+    val VIEW_TYPE_MY = 1
+
+    // 임시의 빈 리스트
+    private var humanList: List<Human> = emptyList()
+
+    override fun getItemViewType(position: Int): Int {
+        return when (humanList[position]) {
+            is Human.Friend -> VIEW_TYPE_FRIEND
+            is Human.My -> VIEW_TYPE_MY
+        }
+    }
```

Comment:
코틀린 문법을 활용한다면 중괄호 내부 return이 아니라 아래처럼 사용할 수도 있겠죠!!
근데 절대 틀린 방법이 아닙니다!! 사람마다 선호하는 방식대로 짜는건데 저는 아래처럼 짜긴 해요 ㅎㅎ
```suggestion
    override fun getItemViewType(position: Int): Int = when (humanList[position]) {
            is Human.Friend -> VIEW_TYPE_FRIEND
            is Human.My -> VIEW_TYPE_MY
        }
```